### PR TITLE
Refactor workflows to add deployment of contracts migrated on Celo

### PR DESCRIPTION
**Pull request by michalinacienciala**
Workflows have been refactored so that Celo-related artifacts could be
deployed in the chain of automatic execusions triggered by the manual
dispatch of the `Main` workflow in the `ci` project (with `environment`
set to `alfajores`).

Similar changes have been introduced to `keep-core` and `tbtc`
workflows. See:
https://github.com/keep-network/keep-core/pull/2492
https://github.com/keep-network/tbtc/pull/804

### CODE REVIEW

File: .github/workflows/client.yml [hunk 1]
```diff
@@ -106,7 +119,7 @@ jobs:
           build-args: |
             REVISION=${{ github.sha }}
           # VERSION= ? TODO: Configure version, sample: 1.7.6
-          push: ${{ github.event_name == 'workflow_dispatch' }}
+          push: true
 
       - # Temp fix - move cache instead of copying (added below step and
```
Comment (nkuba) [inline .github/workflows/client.yml:124]: Can we move it to be the very last step (after `Notify CI about completion of the workflow`)?

**Review by nkuba — CHANGES_REQUESTED** [review]

**Comment by nkuba** [comment]
@michalinacienciala can we solve the conflicts, please?

**Comment by nkuba** [comment]
@michalinacienciala we've got some formatting problems.

**Comment by michalinacienciala** [comment]
> @michalinacienciala we've got some formatting problems.

Fixed in 08bdbe82831b53bd53a2a8c75536aa71bb533cad.

**Comment by michalinacienciala** [comment]
@nkuba, after merging the change, there would be a need for change in the configuration of required jobs - `client-build-test-publish-ethereum` should become required (instead of `client-build-test-publish`).
We should also consider whether to make celo-related jobs required...

**Review by nkuba — CHANGES_REQUESTED** [review]

File: .github/workflows/contracts.yml [hunk 2]
```diff
@@ -167,13 +172,22 @@ jobs:
...
-      - name: Migrate contracts
+      - name: Migrate contracts on Ethereum
+        if: github.event.inputs.environment != 'alfajores'
         env:
-          ETH_HOSTNAME: ${{ secrets.KEEP_TEST_ETH_HOSTNAME }}
-          CONTRACT_OWNER_ETH_ACCOUNT_PRIVATE_KEY: |
+          CHAIN_API_URL: ${{ secrets.KEEP_TEST_ETH_HOSTNAME }}
+          CONTRACT_OWNER_ACCOUNT_PRIVATE_KEY: |
             ${{ secrets.KEEP_TEST_ETH_CONTRACT_OWNER_PRIVATE_KEY }}
         run: npx truffle migrate --reset --network $TRUFFLE_NETWORK
```
[the remark this answers is not in the document]
Comment (michalinacienciala) [reply .github/workflows/contracts.yml:181]: The value is read from either `ropsten.env` or `alfajores.env` config file, thanks to execution of `load-env-variables` action few steps before.
In the `run` parameter both notations work, I think - the `$TRUFFLE_NETWORK` and the `${{ env.TRUFFLE_NETWORK }}`.

File: .github/workflows/contracts.yml [hunk 3]
```diff
@@ -103,20 +103,29 @@ jobs:
...
             ${{ runner.os }}-
 
+      # Below step is a workaround. We have two package-lock.json configs - one
+      # in the `keep-ecdsa` and one in `keep-ecdsa/solidity`.
+      # The `@keep-network/prettier-config-keep` dependency is defined in the
+      # former one, and is not present in the latter one. When `npm run lint` is
+      # run in the `keep-ecdsa/solidity`, the manager looks not only at the
+      # config file in that folder, but also at the config in the root directory.
+      # It throws an error about missing `@keep-network/prettier-config-keep` if
+      # `npm ci` isn't run in the root directory.
```
[the remark this answers is not in the document]
Comment (michalinacienciala) [reply .github/workflows/contracts.yml]: Fixed, see 5ec358d98f04727795ea5052d229a34957d096ef.

File: .github/workflows/contracts.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (nkuba) [reply .github/workflows/contracts.yml:176]: I would do `TRUFFLE_NETWORK: ${{ env.TRUFFLE_NETWORK }}` so it's more clear that we set the value.

[the remark this answers is not in the document]
Comment (michalinacienciala) [reply .github/workflows/contracts.yml]: Done, see 4c988762015b84eb02c69bcbd438de975ce89d6e

**Review by nkuba — APPROVED** [review]

**Pull request merged**